mlir/lib/QEC/Transforms/CliffordTToPPR.cpp

```diff
@@ -130,9 +133,16 @@ LogicalResult controlledConversion(CustomOp op, StringRef P1, StringRef P2,
 // H = (Z · X · Z)π/4
 LogicalResult convertHGate(CustomOp op, ConversionPatternRewriter &rewriter)
 {
-    auto gate = GateConversion({"Z", "X", "Z"}, 4);
-    auto singleQubitOp = singleQubitConversion(op, gate, rewriter);
-    rewriter.replaceOp(op, singleQubitOp);
+    auto Z0 = GateConversion({"Z"}, 4);
+    auto X1 = GateConversion({"X"}, 4);
+    auto Z2 = GateConversion({"Z"}, 4);
+
+    auto types = op.getOutQubits().getType();
+    auto Zppr = singleQubitConversion(op, Z0, rewriter);
+    auto Xppr = singleQubitConversion(Zppr->getLoc(), types, Zppr.getOutQubits(), X1, rewriter);
+    auto Zppr2 = singleQubitConversion(Xppr->getLoc(), types, Xppr.getOutQubits(), Z2, rewriter);
```

Comment:
Are these different function calls necessary? I think the current output is the same if you used the original function instead:
```suggestion
    auto Zppr = singleQubitConversion(op, Z0, rewriter);
    auto Xppr = singleQubitConversion(Zppr, X1, rewriter);
    auto Zppr2 = singleQubitConversion(Xppr, Z2, rewriter);
```

Reply:
Good question! If we pass an operation to that function, the new`ppr` will always be `op.getInputQubit()` (In `getInQubitsAndOutQubits`), while it should be getting `OutQubits` after second `ppr`. This will lead to produce this :
```mlir
%q0 = qec.ppr ["Z"](4) %reg
%q1 = qec.ppr ["X"](4) %reg
%q2 = qec.ppr ["Z"](4) %reg
```

Reply:
Oh you're right, never mind 🤦  The two functions are a bit confusing now, because it's not clear why you would use or the other. Maybe the second version of `singleQubitConversion` could accept a list of GateConversion to apply in a loop, then it would be much clearer how they differ.

Reply:
That's right. It is a bit confusing. 😅 Let me make a change to ensure the functions are clearer.